plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcConfig.java

```diff
@@ -117,9 +150,19 @@ public BaseJdbcConfig setCacheMaximumSize(long cacheMaximumSize)
     @PostConstruct
     public void validate()
     {
-        if (metadataCacheTtl.equals(CACHING_DISABLED) && cacheMaximumSize != BaseJdbcConfig.DEFAULT_METADATA_CACHE_SIZE) {
-            throw new IllegalArgumentException(
-                    format("%s must be set to a non-zero value when %s is set", METADATA_CACHE_TTL, METADATA_CACHE_MAXIMUM_SIZE));
+        if (metadataCacheTtl.equals(CACHING_DISABLED)) {
+            if (cacheMaximumSize != BaseJdbcConfig.DEFAULT_METADATA_CACHE_SIZE) {
+                throw new IllegalArgumentException(
+                        format("%s must be set to a non-zero value when %s is set", METADATA_CACHE_TTL, METADATA_CACHE_MAXIMUM_SIZE));
+            }
+            if (schemaNamesCacheTtl.isPresent() && !schemaNamesCacheTtl.get().equals(CACHING_DISABLED)) {
+                throw new IllegalArgumentException(
+                        format("%s must not be set when %s is not set", METADATA_SCHEMAS_CACHE_TTL, METADATA_CACHE_TTL));
+            }
+            if (tableNamesCacheTtl.isPresent() && !tableNamesCacheTtl.get().equals(CACHING_DISABLED)) {
+                throw new IllegalArgumentException(
+                        format("%s must not be set when %s is not set", METADATA_TABLES_CACHE_TTL, METADATA_CACHE_TTL));
+            }
```

Comment:
nit: Can we expose them as additional methods with `AssertTrue` annotation ? One for schemaName cache and one for tableName cache ?